extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java

```diff
@@ -76,26 +78,33 @@ public List<TransferHistoryEntry> getTransferHistoryEntries() {
         var transferProcesses = getAllTransferProcesses();
 
         return transferProcesses.stream().map(process -> {
-            var agreement = agreementsById.get(process.getDataRequest().getContractId());
-            var negotiation = negotiationsById.get(process.getDataRequest().getContractId());
+            var agreement = Optional.ofNullable(agreementsById.get(process.getDataRequest().getContractId()));
+            var negotiation = Optional.ofNullable(negotiationsById.get(process.getDataRequest().getContractId()));
             var asset = assetLookup(assetsById, process);
-            var direction = ContractAgreementDirection.fromType(negotiation.getType());
+            var direction = negotiation.map(ContractNegotiation::getType).map(ContractAgreementDirection::fromType);
             var transferHistoryEntry = new TransferHistoryEntry();
             transferHistoryEntry.setAssetId(asset.getId());
-            if (direction == ContractAgreementDirection.CONSUMING) {
-                transferHistoryEntry.setAssetName(asset.getId());
-            } else {
-                transferHistoryEntry.setAssetName(
-                        StringUtils.isBlank((String) asset.getProperties().get(Prop.Dcterms.TITLE))
-                                ? asset.getId()
-                                : asset.getProperties().get(Prop.Dcterms.TITLE).toString()
-                );
+
+            if (direction.isPresent()) {
```

Comment:
Did we ensure that the Ui can handle null values for the AssetName? Maybe set the assetName to "" if the value is not present. Or pass an error message.

Reply:
Valid concern.

Testing with missing data:

![image](https://github.com/sovity/edc-extensions/assets/508418/74451d15-91cf-4384-aacb-03184cabd0b1)

![image](https://github.com/sovity/edc-extensions/assets/508418/b665b4f6-1f77-40ed-b53e-eae4f6da58e1)


The UI handles it gracefully. Currently not showing anything but returning null lets the UI decide what and how to show the problem.
If this is the only place where this API is used we're good.

Reply:
Also, no warning nor error was shown in the browser's console.